File: 2013/cracking/01-02.py
# coding=utf8
# Write code to reverse a C-Style String. (C-String means that "abcd" is represented as five characters, including the null character.)

def reverse_string(str):
    buffer = list(str)
    length = len(buffer)
    for i in xrange(length / 2):
        t = buffer[i]
        buffer[i] = str[length - i - 1]
        buffer[length - i - 1] = t
    return "".join(buffer)

print "Result %s" % reverse_string("abcde")

File: 2013/cracking/01-02.c
// Write code to reverse a C-Style String. (C-String means that "abcd" is represented as five characters, including the null character.)

#include <stdio.h>

char* reverse_string(char* str)
{
  char* c = str;
  char* end = str;

  while (*c != '\0')
  {
    end = c++;
  }

  c = str;

  while (c < end)
  {
    char tmp = *c;
    *c = *end;
    *end = tmp;
    c++;
    end--;
  }

  return str;
}

int main() 
{
  char input[] = "Test";
  printf("Reversed string %s \n", reverse_string(input));
  return 0;
}
